packages/pigeon/lib/java_generator.dart

```diff
@@ -14,8 +14,8 @@ const Map<String, String> _javaTypeForDartTypeMap = <String, String>{
   'Int32List': 'int[]',
   'Int64List': 'long[]',
   'Float64List': 'double[]',
-  'List': 'ArrayList',
-  'Map': 'HashMap',
+  'List': 'List<Object>',
+  'Map': 'Map<String, Object>',
```

Comment:
ArrayList and HashMap are directly lifted from the platform channels documentation as the datatypes the standard message codec is creating.  I don't imagine it's safe or possible just to cast from one to the other:

https://flutter.dev/docs/development/platform-integration/platform-channels

Reply:
From what I can see in Flutter's code, they don't care about specific type anywhere:

* https://github.com/flutter/engine/blob/36769af3103819b6e977b71f80761091955fa5d9/shell/platform/android/io/flutter/plugin/common/StandardMessageCodec.java#L260
* https://github.com/flutter/engine/blob/36769af3103819b6e977b71f80761091955fa5d9/shell/platform/android/io/flutter/plugin/common/JSONUtil.java#L71

Reply:
Also I don't see scenario where this conversion code would suddenly start caring about type of Lists. They are just iterating through it and storing it into transfer format.

Reply:
Ok, let me give it a closer look today.  It looks like I'm going to have some time set aside for Pigeon work today.